src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -232,10 +300,20 @@ public ImmutableArray<int> ToImmutableArray()
                 var name = arguments.Names[argumentPosition];
                 for (int p = 0; p < memberParameters.Length; ++p)
                 {
-                    // TODO: p is initialized to zero; is it ok for a named argument to "correspond" to
-                    // _any_ parameter, or just the parameters past the point of positional arguments?
+                    // p is initialized to zero; it is ok for a named argument to "correspond" to
+                    // _any_ parameter (not just the parameters past the point of positional arguments)
                     if (memberParameters[p].Name == name.Identifier.ValueText)
                     {
+                        if (isValidParams && memberParameters[p].IsParams)
```

Comment:
>memberParameters[p].IsParams [](start = 45, length = 28)

This check doesn't feel right. I think we should simply check if this is the last parameter. We could run into a params parameter that is not last, even when the last one is valid params.